OpenUpSA/wazimap-ng
# Notification panel for in progress/completed async task in admin panel

## Description
New notification system for admin panel to check task in progress and completed tasks
Now we will have a bell icon in admin panel header

<img width="553" alt="Screenshot 2020-09-28 at 1 36 32 AM" src="https://user-images.githubusercontent.com/6871866/94374606-1441a500-012b-11eb-9ed0-e521aceae8dc.png">

It will have badge and counter for unread notifications:

<img width="591" alt="Screenshot 2020-09-28 at 1 37 52 AM" src="https://user-images.githubusercontent.com/6871866/94374636-405d2600-012b-11eb-88ba-5179108c9c35.png">

Notifications will marked as unread when user will open the notification panel.
New Notifications will have blue dot in the end.

<img width="400" alt="Screenshot 2020-09-28 at 1 39 21 AM" src="https://user-images.githubusercontent.com/6871866/94374672-76020f00-012b-11eb-8479-e373f744bb25.png">

Notification panel consist of 2 parts:

* Task in progress
* completed tasks

user will be able to clear/remove completed task notifications but task in progress can't be removed.
They will be automatically shifted from in progress to completed when task is finished.

Future Implementations/Improvements:

1) Right now I am soft deleting the notifications user do not wish to see so in future we can have view older notifications to open a link or a panel with older notifications.

2) Data is json field of notifications. I am purposely adding profile id linked to task in data. In future we can also have a panel in which we show user's profile updates done by other user's of same profile
For example user will be able to see/receive notification for uploaded data to a profile by some other admin

3) Using ASGI server to implement notification updates using sockets or django channels.
right now we are sending request to server in every 10 seconds to check for new notifications.

## Related Issue

## How to test it locally

I am having a weird issue on my local.
I am setting `NOTIFICATIONS_NOTIFICATION_MODEL = 'general.Notification'` in common.py config file
Still getting error for duplicate classes.
If you get this error while running on local, Please copy this setting and paste it in your local.py config
It should get it working.

## Changelog

### Added

### Updated

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
